src/env-ssm.ts

```diff
@@ -0,0 +1,185 @@
+  /**
+    * Adds process.env variables to the container (default true).
+    */
+  processEnv?: boolean
+
+  /**
+    * Adds tfvar file variables to the container (default false).
+    */
+  tfvar?: string
+
+  /**
+    * Adds .env file variables to the container (default true).
+    */
+  dotenv?: boolean | string
+}
+export interface ResolvedOptions extends Required<Options> {
+  paths: string[]
+  dotenv: string
+}
+
+export type EnvVar<T extends Extensions = {}> = IEnv<IPresentVariable<T> & ExtenderType<T>, IOptionalVariable<T> & ExtenderTypeOptional<T>>
+
+/**
+ * Coerces input options into a more consistent format and setting defaults
+ */
+async function resolveOptions (options: Options): Promise<ResolvedOptions> {
+  const processEnv = resolveProcessEnv(options)
+  const tfvar = resolveTfVar(options)
+  const dotenv = resolveDotEnv(options)
+  const paths = resolvePaths(options)
+  const ssm = await resolveSSMClient(options)
+  return { processEnv, tfvar, dotenv, ssm, paths }
+}
+
+function resolveProcessEnv (options: Options): boolean {
+  return options.processEnv === undefined ? true : options.processEnv
+}
+
+function resolveTfVar (options: Options): string {
+  return options.tfvar !== undefined ? path.join(process.cwd(), options.tfvar) : ''
+}
+
+function resolveDotEnv (options: Options): string {
+  if (typeof options.dotenv === 'string') {
+    return path.join(process.cwd(), options.dotenv)
+  }
+  return options.dotenv === true ? path.join(process.cwd(), '.env') : ''
+}
+
+function resolvePaths (options: Options): string[] {
+  return !Array.isArray(options.paths)
+    ? [options.paths]
+    : options.paths
+}
+
+async function resolveSSMClient (options: Options): Promise<SSMClient> {
+  // Import ssm client if not provided (optional dependency)
+  return (options.ssm == null)
+    ? new (await import('@aws-sdk/client-ssm')).SSMClient({})
+    : options.ssm
+}
+
+/**
+ * Creates an environment container from a SSM Parameter Store path
+ */
+export default async function EnvSsm (input: string | string[] | Options): Promise<EnvVar> {
+  const options = typeof input === 'string' || Array.isArray(input) ? { paths: input } : input
+  const resolvedOptions = await resolveOptions(options)
+  const { tfvar, dotenv, processEnv } = resolvedOptions
+
+  // Merge all containers in order of precedence: ssm, .env, .tfvar, process.env
+  // Merging with lodash.merge to maintain ssm path tree (e.g. /db/user = 'secret' => {db: {user: 'secret'}})
+  const containers = []
+  if (dotenv !== '') containers.push(loadDotEnv(resolvedOptions))
+  if (tfvar !== '') containers.push(loadTfVar(resolvedOptions))
+  if (processEnv) containers.push(loadProcessEnv())
+  const container = merge(await loadSsmParams(resolvedOptions), ...containers)
+
+  // Ensure all parameter values are of type string and props are uppercase
+  for (const prop in container) {
+    if (!Object.hasOwnProperty.call(container, prop) || typeof container[prop] === 'string') continue
```

Comment:
Why `Object.hasOwnProperty.call(container, prop)` instead of `container.hasOwnProperty(prop)`? I could be wrong about this, but using `call` is may not optimizable for the JIT compiler, or maybe it just runs slower. Of course, it's been a while since I've tried, so it may not run slower any more.

Reply:
Because linters nowadays give you errors like this:
> Do not access Object.prototype method 'hasOwnProperty' from target object             no-prototype-builtins